core/src/components/toolbar/toolbar.ionic.scss

```diff
@@ -61,32 +52,53 @@ $toolbar-order-ionic: (
 
 // Ionic Toolbar Slot Placement
 // --------------------------------------------------
+// We're using the slots to force the main toolbar content to be
+// cenetered in the toolbar. This is a bit of a hack but it works.
+// The main content is placed in the center and then JavaScript evaluates
+// the sizes of the different slots and sets what size the pairs should be
+// based on the larger one. We then use `flex-basis` to set the expected
+// size of the slots and disable `flex-shrink` so that the smaller slot cannot
+// shrink and throw off the center. The slots are paired up so the mirroring slots
+// always have the same size. That is, start and end are paired and primary
+// and secondary are paired. All of this works together to force the main
+// content to maintain the center as best as CSS allows while also allowing
+// the slots and main content to have fairly dynamic widths.
+// --------------------------------------------------
 
-::slotted([slot="secondary"]) {
-  order: map-get($toolbar-order-ionic, slot-secondary);
-}
+:host(.has-end-content) slot[name="end"],
+:host(.show-end) slot[name="end"] {
+  display: flex;
 
-::slotted([slot="primary"]) {
-  order: map-get($toolbar-order-ionic, slot-primary);
+  flex: 1 0 var(--start-end-size, 0);
+  justify-content: flex-end;
 
   text-align: end;
+  gap: globals.$ion-space-400;
 }
 
-::slotted([slot="start"]) {
-  order: map-get($toolbar-order-ionic, slot-start);
+:host(.has-start-content) slot[name="start"],
+:host(.show-start) slot[name="start"] {
+  display: flex;
+
+  flex: 1 0 var(--start-end-size, 0);
+  gap: globals.$ion-space-400;
 }
 
-::slotted([slot="start"]),
-::slotted([slot="end"]) {
+:host(.has-primary-content) slot[name="primary"],
+:host(.show-primary) slot[name="primary"] {
   display: flex;
 
-  flex: 1;
-  gap: globals.$ion-space-200;
+  flex: 1 0 var(--primary-secondary-size, 0);
+  justify-content: flex-end;
+
+  text-align: end;
+  gap: globals.$ion-space-400;
 }
 
-::slotted([slot="end"]) {
-  justify-content: end;
-  order: map-get($toolbar-order-ionic, slot-end);
+:host(.has-secondary-content) slot[name="secondary"],
+:host(.show-secondary) slot[name="secondary"] {
+  display: flex;
 
-  font-size: globals.$ion-font-size-600;
+  flex: 1 0 var(--primary-secondary-size, 0);
+  gap: globals.$ion-space-400;
```

Comment:
```suggestion

```

This gap does nothing, you need to apply it to `ion-buttons`

Reply:
These are an artifact from when slotted content was being forced to flex, I've removed that. I don't think changing ion-buttons is in the scope of this PR